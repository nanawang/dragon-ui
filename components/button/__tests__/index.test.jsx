import React from 'react';
import { render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from '../index';

describe('Button', () => {
  it('renders Buttons of different theme correctly', () => {
    const wrapper = render(
      <div>
        <Button>default</Button>
        <Button theme="primary">info</Button>
        <Button theme="success">success</Button>
        <Button theme="warning">warning</Button>
        <Button theme="error">error</Button>
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders disaled Button correctly', () => {
    const wrapper = render(
      <div>
        <Button disabled>default</Button>
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Buttons with radius or round props correctly', () => {
    const wrapper = render(
      <div>
        <div>
          <Button shape="radius">default</Button>
          <Button shape="radius" theme="primary">info</Button>
          <Button shape="radius" theme="success">success</Button>
          <Button shape="radius" theme="warning">warning</Button>
          <Button shape="radius" theme="danger">error</Button>
        </div>
        <div>
          <Button shape="round">default</Button>
          <Button shape="round" theme="primary">info</Button>
          <Button shape="round" theme="success">success</Button>
          <Button shape="round" theme="warning">warning</Button>
          <Button shape="round" theme="danger">error</Button>
        </div>
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Buttons of different size correctly', () => {
    const wrapper = render(
      <div>
        <Button size="xl" theme="primary">xl尺寸</Button>
        <Button size="lg" theme="primary">lg尺寸</Button>
        <Button theme="primary">默认尺寸</Button>
        <Button size="sm" theme="primary">sm尺寸</Button>
        <Button size="xs" theme="primary">xs尺寸</Button>
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders loading Button correctly', () => {
    const wrapper = render(
      <div>
        <Button theme="info" loading>加载中</Button>
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders anchor Button correctly', () => {
    const wrapper = render(
      <Button href="https://www.baidu.com" theme="primary" target="_blank">百度一下</Button>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders anchor Button with tag a correctly', () => {
    const wrapper = shallow(
      <Button href="https://www.baidu.com">百度一下</Button>
    );
    expect(wrapper.exists('a')).toEqual(true);
  });

  it('triggers onClick callback correctly on nomarl Button', () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <Button onClick={onClick}>加载中</Button>
    );
    wrapper.find('button').simulate('click');
    expect(onClick).toHaveBeenCalled();
  });

  it('should not trigger onClick callback on disabled Button', () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <Button disabled onClick={onClick}>加载中</Button>
    );
    wrapper.find('button').simulate('click');
    expect(onClick).not.toHaveBeenCalled();
  });
});
