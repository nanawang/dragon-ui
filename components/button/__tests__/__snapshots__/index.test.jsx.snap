// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders Buttons of different size correctly 1`] = `
<div>
  <button
    class="is-radius za-button za-button--primary za-button--xl"
    type="button"
  >
    xl尺寸
  </button>
  <button
    class="is-radius za-button za-button--primary za-button--lg"
    type="button"
  >
    lg尺寸
  </button>
  <button
    class="is-radius za-button za-button--primary"
    type="button"
  >
    默认尺寸
  </button>
  <button
    class="is-radius za-button za-button--primary za-button--sm"
    type="button"
  >
    sm尺寸
  </button>
  <button
    class="is-radius za-button za-button--primary za-button--xs"
    type="button"
  >
    xs尺寸
  </button>
</div>
`;

exports[`Button renders Buttons of different theme correctly 1`] = `
<div>
  <button
    class="is-radius za-button za-button--default"
    type="button"
  >
    default
  </button>
  <button
    class="is-radius za-button za-button--primary"
    type="button"
  >
    info
  </button>
  <button
    class="is-radius za-button za-button--success"
    type="button"
  >
    success
  </button>
  <button
    class="is-radius za-button za-button--warning"
    type="button"
  >
    warning
  </button>
  <button
    class="is-radius za-button za-button--error"
    type="button"
  >
    error
  </button>
</div>
`;

exports[`Button renders Buttons with radius or round props correctly 1`] = `
<div>
  <div>
    <button
      class="is-radius za-button za-button--default"
      type="button"
    >
      default
    </button>
    <button
      class="is-radius za-button za-button--primary"
      type="button"
    >
      info
    </button>
    <button
      class="is-radius za-button za-button--success"
      type="button"
    >
      success
    </button>
    <button
      class="is-radius za-button za-button--warning"
      type="button"
    >
      warning
    </button>
    <button
      class="is-radius za-button za-button--danger"
      type="button"
    >
      error
    </button>
  </div>
  <div>
    <button
      class="is-round za-button za-button--default"
      type="button"
    >
      default
    </button>
    <button
      class="is-round za-button za-button--primary"
      type="button"
    >
      info
    </button>
    <button
      class="is-round za-button za-button--success"
      type="button"
    >
      success
    </button>
    <button
      class="is-round za-button za-button--warning"
      type="button"
    >
      warning
    </button>
    <button
      class="is-round za-button za-button--danger"
      type="button"
    >
      error
    </button>
  </div>
</div>
`;

exports[`Button renders anchor Button correctly 1`] = `
<a
  class="is-radius za-button za-button--primary"
  href="https://www.baidu.com"
  target="_blank"
>
  百度一下
</a>
`;

exports[`Button renders disaled Button correctly 1`] = `
<div>
  <button
    class="is-radius is-disabled za-button za-button--default"
    disabled=""
    type="button"
  >
    default
  </button>
</div>
`;

exports[`Button renders loading Button correctly 1`] = `
<div>
  <button
    class="is-radius is-loading za-button za-button--info"
    type="button"
  >
    <i
      class="za-icon za-icon-loading theme-default rotate360"
    />
      加载中
  </button>
</div>
`;
